src/plans/D3MCompoundPlan.sol

```diff
@@ -0,0 +1,198 @@
+interface InterestRateModelLike {
+    function baseRatePerBlock()       external view returns (uint256);
+    function kink()                   external view returns (uint256);
+    function multiplierPerBlock()     external view returns (uint256);
+    function jumpMultiplierPerBlock() external view returns (uint256);
+}
+
+contract D3MCompoundPlan is ID3MPlan {
+
+    mapping (address => uint256) public wards;
+    InterestRateModelLike        public tack;
+    address                      public delegate; // cDai implementation
+    uint256                      public barb;     // target Interest Rate Per Block [wad] (0)
+
+    CErc20Like public immutable cDai;
+
+    // https://github.com/compound-finance/compound-protocol/blob/a3214f67b73310d547e00fc578e8355911c9d376/contracts/CTokenInterfaces.sol#L31
+    uint256 internal constant MAX_BORROW_RATE = 0.0005e16;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address cDai_) {
+        address rateModel_ = CErc20Like(cDai_).interestRateModel();
+        address delegate_  = CErc20Like(cDai_).implementation();
+
+        require(rateModel_ != address(0), "D3MCompoundPlan/invalid-rateModel");
+        require(delegate_  != address(0), "D3MCompoundPlan/invalid-delegate");
+
+        cDai     = CErc20Like(cDai_);
+        tack     = InterestRateModelLike(rateModel_);
+        delegate = delegate_;
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MCompoundPlan/not-authorized");
+        _;
+    }
+
+    // --- Math ---
+    uint256 internal constant WAD = 10 ** 18;
+    function _wmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * y) / WAD;
+    }
+    function _wdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = (x * WAD) / y;
+    }
+
+    // --- Admin ---
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "barb") {
+            require(data <= MAX_BORROW_RATE, "D3MCompoundPlan/barb-too-high");
+            barb = data;
+        } else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+    function file(bytes32 what, address data) external auth {
+        if (what == "tack") tack = InterestRateModelLike(data);
+        else if (what == "delegate") delegate = data;
+        else revert("D3MCompoundPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
```

Comment:
nit: can set the same order than aave plan where first is `_calculateTargetSupply` than `getTargetAssets` ?

Reply:
There are also minor differences in `getTargetAssets` that could be exact same code for both, like spacing, order, use of parenthesis or not. If we could standardize it for at least these two is much better when doing diffs.
Maybe the Compound one is better and we should just change in Aave one.

Reply:
Hmm I think we defined for this repo to put all the internals before. In fact the Aave plan doesn't follow it correctly as was defined for the Hub neither. But I definitely wouldn't take what was in DssVest as a defined standard globally.
So I'd we come at least for some standard within this repo and in this case that Compound and Aave are very similar try to keep the code differences the minimum as possible. For both `_calculateTargetSupply` should be moved right after internal math functions.

Reply:
I'll take care of it in `v2` branch so then you can pull mirror both the best is possible.

Reply:
I was actually confused, in Hub we have first the administration external functions than the internals which is actually the same than the actual Aave plan. I'd say then, for now we stick to that, so nothing to be done in `v2`.

Reply:
@talbaneth, I confirm this was discussed on `v2` and the general consensus was following a different code layout pattern for d3m, diverging from `dss-vest` and the canonical in solc docs, grouping internal functions before targeted external ones (aka Gonza's standard) IIRC.

Reply:
Ok sounds good, we didn't discuss that but we can do in these days. If Compund way is better we can change the Aave one.

Reply:
Did some changes in V2 and here to better align code style.
Not everything is fully aligned but I think diffing is easier now.